p4a: compile against non GPL ffmpeg version

I try to compile ffpyplayer against non GPL ffmpeg with p4a. I use version n4.3.1 of ffmpeg and v4.3.1 of ffpyplayer.

ffpyplayer build fails during p4a run due to missing ``postprocess.h``, which is not included in built when compiling ffmpeg without GPL components.

>   In file included from ffpyplayer/pic.c:653:	
    ffpyplayer/clib/misc.h:18:10: fatal error: 'libpostproc/postprocess.h' file not	
          found	
    #include "libpostproc/postprocess.h"	
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~

Any ideas?

After setting CONFIG_POSTPROC=0 in the ffpyplayer recipe the compiler not complains any more, but now the linker fails with

  > aarch64-linux-android/bin/ld: cannot find -lpostproc	
clang: error: linker command failed with exit code 1

I tried to exclude from libraries here https://github.com/matham/ffpyplayer/compare/master...rnixx:disable_postproc?expand=1 but it has no effect.

Where are the libraries to link defined?

--- a/setup.py
+++ b/setup.py
@@ -4,6 +4,7 @@ from distutils.command.build_ext import build_ext
 import sys
 import ffpyplayer
 
+
 try:
     from setuptools import setup, Extension
     print('Using setuptools')
@@ -206,8 +207,10 @@ elif "NDKPLATFORM" in environ:
     ffmpeg_lib, ffmpeg_include = get_paths('FFMPEG')
     libraries.extend([
         'avcodec', 'avdevice', 'avfilter', 'avformat',
-        'avutil', 'swscale', 'swresample', 'postproc', 'm'
+        'avutil', 'swscale', 'swresample', 'm'
     ])
+    if environ.get('CONFIG_POSTPROC') != '0':
+        libraries.extend(['postproc'])
     library_dirs.append(ffmpeg_lib)
     include_dirs.append(ffmpeg_include)
 
@@ -234,8 +237,12 @@ elif "NDKPLATFORM" in environ:
 else:
 
     # ffmpeg
-    objects = ['avcodec', 'avdevice', 'avfilter', 'avformat',
-                   'avutil', 'swscale', 'swresample', 'postproc']
+    objects = [
+        'avcodec', 'avdevice', 'avfilter', 'avformat',
+        'avutil', 'swscale', 'swresample'
+    ]
+    if environ.get('CONFIG_POSTPROC') != '0':
+        objects.extend(['postproc'])
     for libname in objects[:]:
         for key, val in c_options.items():
             if key.endswith(libname) and not val:
@@ -417,4 +424,3 @@ setup(name='ffpyplayer',
       data_files=get_wheel_data(),
       cmdclass=cmdclass, ext_modules=ext_modules,
       setup_requires=setup_requires)
-

--- a/setup.py
+++ b/setup.py
@@ -209,8 +209,8 @@ elif "NDKPLATFORM" in environ:
         'avcodec', 'avdevice', 'avfilter', 'avformat',
         'avutil', 'swscale', 'swresample', 'm'
     ])
-    if environ.get('CONFIG_POSTPROC') != '0':
-        libraries.extend(['postproc'])
+    if c_options['config_postproc']:
+        libraries.append('postproc')
     library_dirs.append(ffmpeg_lib)
     include_dirs.append(ffmpeg_include)
 
@@ -241,8 +241,8 @@ else:
         'avcodec', 'avdevice', 'avfilter', 'avformat',
         'avutil', 'swscale', 'swresample'
     ]
-    if environ.get('CONFIG_POSTPROC') != '0':
-        objects.extend(['postproc'])
+    if c_options['config_postproc']:
+        objects.append('postproc')
     for libname in objects[:]:
         for key, val in c_options.items():
             if key.endswith(libname) and not val:

